src/foam/nanos/crunch/predicate/IsAgent.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.predicate',
+  name: 'IsAgent',
+
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  implements: ['foam.core.Serializable'],
+
+  documentation: `
+    Returns true if user and realUser are different.
+    This is useful for agent association capabilities.
+  `,
+
+  javaImports: [
+    'foam.core.X',
+    'foam.nanos.auth.Subject'
+  ],
+
+  methods: [
+    {
+      name:'f',
+      javaCode: `
+        if ( ! ( obj instanceof X ) ) return false;
+        X x = (X) obj;
+        return ((Subject) x.get("subject")).getUser().getId()
+          != ((Subject) x.get("subject")).getRealUser().getId();
```

Comment:
isAgent might make a good method on the subject itself.

Reply:
I agree with this. tagging @kgrgreer @penzital in case they have comments.

Reply:
Agreed.

Reply:
Updated